src/bwt_util.rs

```diff
@@ -1,3 +1,111 @@
+use bitvec::prelude::*;
+use itertools::Itertools;
+use std::clone::Clone;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::hash::Hash;
+
+/// implementation of the mergeIter function as described in Holt & McMillan 2014
+/// Takes the current state of the interleave bitvector, the two burrows wheeler
+/// transform slices, and a mapping from character to number of lexigraphically
+/// smaller characters in the BWTs provided and does a round of interleave
+/// resolution
+pub fn pairwise_merge_iter<T: Ord + Hash + Clone + Debug>(
+    current_interleave: &BitVec<u64, Msb0>,
+    bwt0: &[T],
+    bwt1: &[T],
+    offsets: &HashMap<T, usize>,
+) -> BitVec<u64, Msb0> {
+    // initial conditions
+    let mut next_interleave = bitvec![u64, Msb0; 0; current_interleave.len()];
+    let mut current_pos0 = 0usize;
+    let mut current_pos1 = 0usize;
+
+    let mut temp_index = offsets.clone();
+
+    // iterate through each bit
+    for b in current_interleave.iter() {
+        let c = if *b {
+            let v = bwt0[current_pos0].clone();
```

Comment:
I think these clones can also be removed, making it a `let c: &T = if ...`.  So you'll have to update the `&c` to just `c` below, but should remove any unnecessary allocs.